kernel/kernel_drivers.go

```diff
@@ -157,27 +158,29 @@ func createModulesSubtree(kernelMount, kernelTree, kversion, kname string, krev
 	}
 
 	// If necessary, add modules from components and run depmod
-	return setupModsFromComp(kernelTree, kversion, kname, krev, compInfos)
+	return setupModsFromComp(kernelTree, kversion, kmodsConts)
 }
 
-func setupModsFromComp(kernelTree, kversion, kname string, krev snap.Revision, compInfos []*snap.ComponentSideInfo) error {
+func setupModsFromComp(kernelTree, kversion string, kmodsConts []snap.ContainerPlaceInfo) error {
 	// This folder needs to exist always to allow for directory swapping
 	// in the future, even if right now we don't have components.
 	compsRoot := filepath.Join(kernelTree, "lib", "modules", kversion, "updates")
 	if err := os.MkdirAll(compsRoot, 0755); err != nil {
 		return err
 	}
 
-	if len(compInfos) == 0 {
+	if len(kmodsConts) == 0 {
 		return nil
 	}
 
 	// Symbolic links to components
-	for _, ci := range compInfos {
-		compPI := snap.MinimalComponentContainerPlaceInfo(ci.Component.ComponentName,
-			ci.Revision, kname)
-		lname := filepath.Join(compsRoot, ci.Component.ComponentName)
-		to := filepath.Join(compPI.MountDir(), "modules", kversion)
+	for _, kmc := range kmodsConts {
+		snapComp := strings.Split(kmc.ContainerName(), "+")
```

Comment:
this looks a bit strange, should it be a helper in snap? should we continue to pass something a bit richer to this?

Reply:
I have added a helper in naming in the end.